consensus/polybft/blockchain_wrapper.go

```diff
@@ -0,0 +1,191 @@
+	return p.blockchain.Header()
+}
+
+// CommitBlock commits a block to the chain
+func (p *blockchainWrapper) CommitBlock(stateBlock *StateBlock) error {
+	// logs := buildLogsFromReceipts(stateBlock.Receipts, stateBlock.Block.GetHeader())
+	// status, err := p.blockchain.WriteBlockAndSetHead(stateBlock.Block,
+	// stateBlock.Receipts, logs, stateBlock.State, true)
+	// if err != nil {
+	// 	return err
+	// } else if status != core.CanonStatTy {
+	// 	return fmt.Errorf("non canonical change")
+	// }
+	return p.blockchain.WriteBlock(stateBlock.Block, "consensus")
+}
+
+// ProcessBlock builds a final block from given 'block' on top of 'parent'
+func (p *blockchainWrapper) ProcessBlock(parent *types.Header, block *types.Block) (*StateBlock, error) {
+	// TODO: Call validate block in polybft
+	header := block.Header.Copy()
+
+	//transition, err := p.executor.BeginTxn(parent.Hash, header, b.params.Coinbase)
+	transition, err := p.executor.BeginTxn(parent.StateRoot, header, types.Address{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Nemanja - apply transactions from block
+
+	_, root := transition.Commit()
+
+	if root != block.Header.StateRoot {
+		return nil, fmt.Errorf("incorrect state root: (%s, %s)", root, block.Header.StateRoot)
+	}
+
+	// build the block
+	found := consensus.BuildBlock(consensus.BuildBlockParams{
+		Header:   header,
+		Txns:     block.Transactions,
+		Receipts: transition.Receipts(),
+	})
+
+	return &StateBlock{
+		Block:    found,
+		Receipts: transition.Receipts(),
+		State:    transition,
+	}, nil
+}
+
+// GetStateProviderForBlock is an implementation of blockchainBackend interface
+func (p *blockchainWrapper) GetStateProviderForBlock(header *types.Header) (contract.Provider, error) {
+	transition, err := p.executor.BeginTxn(header.StateRoot, header, types.ZeroAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewStateProvider(transition), nil
+}
+
+// GetStateProvider returns a reference to make queries to the provided state
+func (p *blockchainWrapper) GetStateProvider(transition *state.Transition) contract.Provider {
+	return NewStateProvider(transition)
+}
+
+// GetHeaderByNumber is an implementation of blockchainBackend interface
+func (p *blockchainWrapper) GetHeaderByNumber(number uint64) (*types.Header, bool) {
+	return p.blockchain.GetHeaderByNumber(number)
+}
+
+// GetHeaderByHash is an implementation of blockchainBackend interface
+func (p *blockchainWrapper) GetHeaderByHash(hash types.Hash) (*types.Header, bool) {
+	return p.blockchain.GetHeaderByHash(hash)
+}
+
+// NewBlockBuilder is an implementation of blockchainBackend interface
+func (p *blockchainWrapper) NewBlockBuilder(
+	parent *types.Header, coinbase types.Address) (blockBuilder, error) {
+	return NewBlockBuilder(&BlockBuilderParams{
+		Parent:      parent,
+		Coinbase:    coinbase,
+		ChainConfig: p.blockchain.Config(),
+		//ChainContext:  p.blockchain,
+		//TxPoolFactory: blockbuilder.NewEthTxPool(p.eth.TxPool()),
```

Comment:
Leftover?

Reply:
This is changed in another pr https://github.com/0xPolygon/polygon-edge/pull/789